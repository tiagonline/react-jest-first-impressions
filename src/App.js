import React, { Component } from 'react';
import {Provider} from 'react-redux';

import store from './store';

import Todos from './Todos';

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <div className="App">
         <h1>Hello World</h1>  
         <Todos />
        </div>
      </Provider>
    );
  }
}

export default App;
