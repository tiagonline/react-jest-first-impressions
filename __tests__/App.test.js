import React from 'React';
import { shallow } from 'enzyme';
import App from '../src/App';

describe('Testing App component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <App />
        );
        expect(wrapper).toMatchSnapshot();
    });
});