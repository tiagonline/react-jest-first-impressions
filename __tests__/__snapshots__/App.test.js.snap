// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Hello World
</h1>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello World",
      },
      "ref": null,
      "rendered": "Hello World",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Hello World
</h1>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello World",
        },
        "ref": null,
        "rendered": "Hello World",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
